src/vos/vos_aggregate.c

```diff
@@ -847,14 +1159,14 @@ insert_segments(daos_handle_t ih, struct agg_merge_window *mw,
 			continue;
 		}
 
-		mark_yield(&phy_ent->pe_addr, acts);
-
-		rc = evt_delete(oiter->it_hdl, &rect, NULL);
-		if (rc) {
-			D_ERROR("Delete "DF_RECT" pe_off:"DF_U64" error: "
+		if (!phy_ent->pe_csum_info.cs_not_valid) {
```

Comment:
Here is the problem I mentioned: a good physical entry is overlapping with a bad physical entry, here we deleted the good physical entry and keep the bad one unchanged. I think we'd:
- either adjust the segment in fill_one_segment() to exclude bad logical part, or;
- abort current evtree aggregation;

Reply:
I keep every physical extent for that output segment, not just the ones that had the error, and I don't insert the logical segment.   Hence, I don't delete the overlapping entry.

I agree that it is confusing that I set the state in each csum info.  It's the only struct I have available when I make the determination, since I put the agg_phy_ent back as private in  vos_aggregation.c

Reply:
Following our Skype discussion, I moved the flag to the agg_phy_ent struct (bool pe_retain) and I set it in vos_aggregation.